keps/sig-node/3953-node-resource-hot-plug/README.md

```diff
@@ -0,0 +1,704 @@
+- Inaccurate resource allocation during cluster initialization.
+- Escalating workload over time, necessitating supplementary resources within the cluster.
+
+To address these situations, we can:
+- Horizontally scale the cluster by incorporating additional compute nodes.
+- Vertically scale the cluster by augmenting node capacity. Currently, the method to capture node resizing within the cluster entails restarting the Kubelet.
+
+These strategies enable the cluster to adapt to varying resource demands, ensuring optimal performance and efficient resource utilization. However, the limitation of requiring a Kubelet restart for node resizing is an area for potential improvement.
+
+Node resource hot plugging offers benefits in situations like:
+- Managing resource demand with a restricted number of nodes by enhancing the capacity of current nodes rather than creating new ones.
+- The process of creating new nodes is more time-consuming compared to augmenting the capacity of existing nodes.
+ 
+This approach allows for more efficient resource management and quicker capacity adjustments, optimizing the utilization of existing hardware.
+### Goals
+
+* Achieve seamless node capacity expansion through hot plugging resources, all without necessitating a kubelet restart.
+* Facilitate the reinitialization of resource managers (CPU manager, memory manager) to accommodate alterations in the node's resource allocation.
+
+### Non-Goals
+
+* Dynamically adjust system reserved and kube reserved values.
+* Hot unplug of node resources.
+* Update the autoscaler to utilize resource hot plugging.
+
+## Proposal
+
+
+This KEP strives to enable node resource hot plugging by incorporating a polling mechanism within the kubelet to retrieve machine-information from cAdvisor's cache, which is already updated periodically.
+The kubelet will periodically fetch this information, subsequently entrusting the node status updater to disseminate these updates at the node level across the cluster.
+Moreover, this KEP aims to refine the initialization and reinitialization processes of resource managers, including the memory manager and CPU manager, to ensure their adaptability to changes in node configurations.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, I must be able to increase the cluster resource capacity without adding a new node to the cluster.
+
+#### Story 2
+
+As a cluster admin, I must be able to increase the cluster resource capacity without need to restart the kubelet.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+1. Node resource hot plugging is an opt-in feature, merging the
+   feature related changes won't impact existing workloads. Moreover, the feature
+   will be rolled out gradually, beginning with an alpha release for testing and
+   gathering feedback. This will be followed by beta and GA releases as the
+   feature matures and potential problems and improvements are addressed.
```

Comment:
while this is true, this is technically true for every change governed by the KEP process.

Do we have more risks specific to this change to elaborate?
- code complixity?
- possible coordination (or lack thereof) with NRI plugins/runtime?

Reply:
Updated the risk and mitigation section please take a look, Thanks.